scripts/getVSCodeExtensionManifest.ts

```diff
@@ -0,0 +1,179 @@
+          title: `Xata: ${command.title}`,
+        };
+
+        if (command.icon) paletteCommand.icon = `$(${command.icon})`;
+
+        contributesCommands.push(paletteCommand);
+        if (command.paletteCondition) {
+          switch (command.paletteCondition) {
+            case "isLogged":
+              contributeMenusCommandPalette.push({
+                command: commandId,
+                when: "xata.isLogged",
+              });
+              break;
+            case "isNotLogged":
+              contributeMenusCommandPalette.push({
+                command: commandId,
+                when: "!xata.isLogged",
+              });
+              break;
+            default:
+              throw new Error("this condition is not handled");
+          }
+        }
+      }
+    }
+
+    // StandAlone command
+    if (command.type === "standAlone") {
+      contributesCommands.push({
+        command: `xata.${command.id}`,
+        title: command.title,
+      });
+      hideFromPalette(`xata.${command.id}`);
+    }
+
+    // TreeItem command
+    if (command.type === "treeItem") {
+      const commandId = `xata.${command.id}`;
+      const treeItemCommand: ContributesCommands[-1] = {
+        command: commandId,
+        title: command.title,
+      };
+
+      if (command.icon) treeItemCommand.icon = `$(${command.icon})`;
+
+      contributesCommands.push(treeItemCommand);
+      hideFromPalette(`xata.${command.id}`);
+
+      command.contexts.forEach((context) => {
+        if (context.item === "workspaceNavigationItem") return;
+
+        const menuItem: MenuItemContext[-1] = {
+          command: commandId,
+          when: `view == ${context.view} && viewItem == ${context.item}`,
+        };
+
+        if (context.group) menuItem.group = context.group;
+
+        contributeMenusItemContext.push(menuItem);
+      });
+    }
+  });
+
+  return {
+    contributes: {
+      commands: contributesCommands,
+      menus: {
+        commandPalette: contributeMenusCommandPalette,
+        "view/item/context": contributeMenusItemContext,
+        "view/title": [
+          {
+            command: "xata.refresh",
+            when: "view in xata.treeViews",
+            group: "navigation",
+          },
+          {
+            command: "xata.addWorkspace",
+            when: "view == xataExplorer && xata.isLogged",
+          },
+          {
+            command: "xata.logout",
+            when: "view == xataExplorer && xata.isLogged",
+          },
+          {
+            command: "xata.login",
+            when: "view == xataExplorer && !xata.isLogged",
+          },
+          {
+            command: "xata.addTable",
+            when: "view == xataWorkspace && workspaceFolderCount == 1 && xata.configState == logged",
+            group: "navigation",
+          },
+          {
+            command: "xata.createBranch",
+            when: "view == xataWorkspace && workspaceFolderCount == 1 && xata.configState == logged",
```

Comment:
This could be extracted to be part of the command manifest (to be closer of the command), but since this is small and this PR was already long, I decide to keep it for later